packages/paymaster/src/BiconomyPaymaster.ts

```diff
@@ -1,49 +1,241 @@
+  ): Promise<BiconomyTokenPaymasterFeeQuoteResponse | string> {
+    userOp = await resolveProperties(userOp)
+    userOp.nonce = BigNumber.from(userOp.nonce).toHexString()
+    userOp.callGasLimit = BigNumber.from(userOp.callGasLimit).toHexString()
+    userOp.verificationGasLimit = BigNumber.from(userOp.verificationGasLimit).toHexString()
+    userOp.maxFeePerGas = BigNumber.from(userOp.maxFeePerGas).toHexString()
+    userOp.maxPriorityFeePerGas = BigNumber.from(userOp.maxPriorityFeePerGas).toHexString()
+    userOp.preVerificationGas = BigNumber.from(userOp.preVerificationGas).toHexString()
+    userOp.signature = '0x'
+    userOp.paymasterAndData = '0x'
+
+    let mode = null
+    let preferredToken = null
+    let feeTokensArray: string[] = []
+    // could make below null
+    let smartAccountInfo = {
+      name: 'BICONOMY',
+      version: '1.0.0'
+    }
+    let webhookData = null
+
+    if (paymasterServiceData.mode) {
+      Logger.log('Requested mode is ', paymasterServiceData.mode)
+      mode = paymasterServiceData.mode
+      // Validation on the mode passed / define allowed enums
+    }
+
+    if (paymasterServiceData.tokenInfo) {
+      if (paymasterServiceData.tokenInfo.preferredToken) {
+        Logger.log('passed preferred token is ', paymasterServiceData.tokenInfo.preferredToken)
+        preferredToken = paymasterServiceData.tokenInfo.preferredToken
+      }
+    }
+
+    Logger.log('userop is ', userOp)
+    // userOp = hexifyUserOp(userOp)
+
+    if (
+      paymasterServiceData.tokenInfo &&
+      paymasterServiceData.tokenInfo.tokenList &&
+      paymasterServiceData.tokenInfo.tokenList.length != 0
+    ) {
+      feeTokensArray = paymasterServiceData.tokenInfo.tokenList
+    }
+
+    const feeQuotes: Array<PaymasterFeeQuote> = []
+
+    if (paymasterServiceData.sponsorshipInfo) {
+      if (paymasterServiceData.sponsorshipInfo.webhookData) {
+        webhookData = paymasterServiceData.sponsorshipInfo.webhookData
+      }
+
+      // Could check here that this must be provided
+      if (paymasterServiceData.sponsorshipInfo.smartAccountInfo) {
+        smartAccountInfo = paymasterServiceData.sponsorshipInfo.smartAccountInfo
+      }
+    }
+
+    try {
+      const response: PaymasterServiceSuccessResponse = await sendRequest({
+        url: `${this.paymasterConfig.paymasterUrl}`,
+        method: HttpMethod.Post,
+        body: {
+          method: 'pm_getFeeQuoteOrData',
+          params: [
+            userOp,
+            {
+              ...(mode !== null && { mode }), // Review can be dynamic
+              tokenInfo: {
+                tokenList: feeTokensArray,
+                ...(preferredToken !== null && { preferredToken })
+              },
+              sponsorshipInfo: {
+                ...(webhookData !== null && { webhookData }),
+                smartAccountInfo: smartAccountInfo
+              }
+            }
+          ], // As per current API
+          id: 4337,
```

Comment:
@tomarsachin2271 suggested using 'currentTimeStamp' as a unique ID for each request. You can utilize the 'getTimestampInSeconds()' function from the common package to generate a timestamp to use as the ID.

Reply:
yup done